testing/junit/src/main/java/com/linecorp/armeria/testing/junit/server/mockwebserver/MockResponseBuilder.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.testing.junit.server.mockwebserver;
+
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+
+/**
+ * A builder of {@link MockResponse}. Obtain a new builder by calling
+ * {@link MockResponse#builder(AggregatedHttpResponse)}.
+ */
+public class MockResponseBuilder {
+
+    private final AggregatedHttpResponse response;
+
+    private Duration headersDelay = Duration.ZERO;
+    private Duration bodyDelay = Duration.ZERO;
+    private Duration trailersDelay = Duration.ZERO;
+
+    MockResponseBuilder(AggregatedHttpResponse response) {
+        this.response = response;
+    }
+
+    /**
+     * Sets a time to wait before returning response headers. Can be useful to validate client timeout behavior.
+     */
+    public MockResponseBuilder headersDelay(Duration headersDelay) {
+        this.headersDelay = headersDelay;
+        return this;
+    }
+
+    /**
+     * Sets a time to wait before returning response headers. Can be useful to validate client timeout behavior.
+     */
+    public MockResponseBuilder headersDelay(int amount, TimeUnit timeUnit) {
+        this.headersDelay = Duration.ofMillis(timeUnit.toMillis(amount));
+        return this;
+    }
+
+    /**
+     * Sets a time to wait before returning response content. Can be useful to validate client timeout behavior.
+     * This value does not affect the time that headers are returned, to set that use
+     * {@link #headersDelay(Duration)}.
+     */
+    public MockResponseBuilder contentDelay(Duration bodyDelay) {
+        this.bodyDelay = bodyDelay;
+        return this;
+    }
+
+    /**
```

Comment:
rather than repeat "validate client timeout behaviour" I would use an example relevant to the method. ex different text for headers, content and trailers ex one use case. I'm not sure that many people are even aware of trailers existence so that is probably an especially good one to note.

Reply:
I ended up with these three methods copying the API from okhttp and adding trailers for symmetry. But I can't think of a use case to separate them so don't know how to do that - instead just removed the knobs to have a single delay which should be plenty.